### PR TITLE
Add support for message-tags

### PR DESCRIPTION
**Pull request by progval**
Not doing anything with it yet, though.

If that's fine with you, I have another PR ready, to use [server-time](https://ircv3.net/specs/extensions/server-time)

### CODE REVIEW

**Comment by SilverRainZ** [comment]
I think we should check the boundary of `current_tag_{key,value}_ptr` at every time we bump the value of them, Otherwise, we may overflow our stack when encountering malformed messages.

**Comment by SilverRainZ** [comment]
Use `strtok` is safer than `ptr++`.

File: src/sirc/sirc_parse.c [hunk 1]
```diff
@@ -80,20 +95,118 @@ SircMessage* sirc_parse(char *line){
...
+        size_t i=0;
+        char current_tag_key[TAGS_SIZE_LIMIT];
+        char current_tag_value[TAGS_SIZE_LIMIT];
+        char *current_tag_key_ptr = current_tag_key;
+        char *current_tag_value_ptr = current_tag_value;
+        gboolean in_key = TRUE;
+        for (char *p=tags_ptr; ; p++){
+            if (*p == '\0') {
+                ERR_FR("Unexpected null byte in message tags");
+                goto bad;
```
[the remark this answers is not in the document]
Comment (progval) [reply src/sirc/sirc_parse.c:132]: Isn't `imsg` freed in `bad`?

[the remark this answers is not in the document]
Comment (SilverRainZ) [reply src/sirc/sirc_parse.c:132]: Yes, my fault.

**Comment by SilverRainZ** [comment]
Sorry for delay again, I am quite busy with both my life and work.

I will test the function of this PR weekend, then evently  it can be merged.

**Review by SilverRainZ — APPROVED** [review]

**Pull request merged**